### PR TITLE
Add Adelais PCB. Adelais RGB rev.3, Adelais rev. 4 APM32F103, Adelais AVR rev. 1

### PR DESCRIPTION
**Pull request by mechlovin**
Add Adelais RGB_LED rev. 3 (STM32F303)
Adelais rev.4 APM32F103
Adelais AVR rev. 1 (Atmega32u4)

## Description

Add Adelais PCB.
Adelais En Ciel Rev. 3: add Adelais RGB Rev. 3, adelais/rgb_led/rev3 (STM32F303 - IS31FL3741)
Adelais rev. 4: add support for APM32F303CBT6, adelais/standard_led/arm/rev4/apm32f103
Adelais AVR: add Adelais AVR rev. 1 (Atmega32u4), adelais/standard_led/avr/rev1.
Thank you.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mechlovin/adelais/rgb_led/rev1/config.h [hunk 1]
```diff
@@ -13,7 +13,12 @@
 #define RGB_MATRIX_KEYRELEASES // reacts to keyreleases (instead of keypresses)
 #define RGB_MATRIX_FRAMEBUFFER_EFFECTS
 #define RGB_DISABLE_AFTER_TIMEOUT 0 // number of ticks to wait until disabling effects
-// #define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
+#define RGB_DISABLE_WHEN_USB_SUSPENDED false // turn off effects when suspended
```
Comment (drashna) [inline keyboards/mechlovin/adelais/rgb_led/rev1/config.h:16]:
```suggestion
#define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
```

File: keyboards/mechlovin/adelais/rgb_led/rev2/config.h [hunk 2]
```diff
@@ -31,7 +31,7 @@
 #define RGB_MATRIX_KEYRELEASES // reacts to keyreleases (instead of keypresses)
 #define RGB_MATRIX_FRAMEBUFFER_EFFECTS
 #define RGB_DISABLE_AFTER_TIMEOUT 0 // number of ticks to wait until disabling effects
-// #define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
+#define RGB_DISABLE_WHEN_USB_SUSPENDED false // turn off effects when suspended
```
Comment (drashna) [inline keyboards/mechlovin/adelais/rgb_led/rev2/config.h:34]:
```suggestion
#define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
```

File: keyboards/mechlovin/adelais/rgb_led/rev3/config.h [hunk 3]
```diff
@@ -0,0 +1,51 @@
+#pragma once
```
Comment (drashna) [inline keyboards/mechlovin/adelais/rgb_led/rev3/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/mechlovin/adelais/rgb_led/rev3/config.h [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+#define PRODUCT      Adelais En Ciel Rev3
+
+#define MATRIX_ROW_PINS { A4, A3, C13, A1, A2}
+#define MATRIX_COL_PINS { B10, B11, B2, B1, B0, A6, A5, A14, A13, A15, B3, B4, B5, B8, B9 }
+
+#define RGB_DI_PIN A7
+#define WS2812_SPI SPID1 // default: SPID1
+#define WS2812_SPI_MOSI_PAL_MODE 5 // Pin "alternate function", see the respective datasheet for the appropriate values for your MCU. default: 5
+#define RGBLED_NUM      23
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/mechlovin/adelais/rgb_led/rev3/config.h:13]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/mechlovin/adelais/rgb_led/rev3/config.h [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+#define DRIVER_ADDR_1 0b0110010
+#define DRIVER_ADDR_2 0b0110010
+#define DRIVER_COUNT 1
+#define DRIVER_1_LED_TOTAL 66
+#define DRIVER_LED_TOTAL DRIVER_1_LED_TOTAL
+#define RGB_MATRIX_KEYPRESSES // reacts to keypresses
+#define RGB_MATRIX_KEYRELEASES // reacts to keyreleases (instead of keypresses)
+#define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+#define RGB_DISABLE_AFTER_TIMEOUT 0 // number of ticks to wait until disabling effects
+#define RGB_DISABLE_WHEN_USB_SUSPENDED false // turn off effects when suspended
```
Comment (drashna) [inline keyboards/mechlovin/adelais/rgb_led/rev3/config.h:35]:
```suggestion
#define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
```

File: keyboards/mechlovin/adelais/rgb_led/rev3/rev3.c [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "rev3.h"
+
+#ifdef RGB_MATRIX_ENABLE
+const is31_led g_is31_leds[DRIVER_LED_TOTAL] = {
```
Comment (drashna) [inline keyboards/mechlovin/adelais/rgb_led/rev3/rev3.c:20]:
```suggestion
const is31_led __flash g_is31_leds[DRIVER_LED_TOTAL] = {
```

File: keyboards/mechlovin/adelais/rgb_led/rev3/rev3.c [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+    }, {
+        1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1,            1,           1,           1,
+        1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1,            1,           1,           1,  
+        1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1,                         1,           1,  
+        1,                      1,          1,          1,          1,          1,          1,          1,          1,          1,          1,            1,           1,           1,  
+        1,                      1,                      1,                      1,                      1,                      1,          1,            1, 
+    } 
+};
+
+void rgb_matrix_indicators_user(void) {
```
Comment (drashna) [inline keyboards/mechlovin/adelais/rgb_led/rev3/rev3.c:115]:
```suggestion
__attribute__((weak)) void rgb_matrix_indicators_user(void) {
```

File: keyboards/mechlovin/adelais/rules.mk [hunk 8]
```diff
@@ -8,7 +8,7 @@ BOOTLOADER = stm32-dfu
 # Build Options
 #   change yes to no to disable
 #
-BOOTMAGIC_ENABLE = lite     # Enable Bootmagic Lite
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/mechlovin/adelais/rules.mk:11]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/mechlovin/adelais/standard_led/avr/rev1/config.h [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+#define LED_PIN_ON_STATE 1
+
+#define BACKLIGHT_PIN B5
+#define BACKLIGHT_LEVELS 3
+#define BACKLIGHT_BREATHING
+
+#define RGB_DI_PIN E2
+#define RGBLED_NUM 23
+#define RGBLIGHT_LIMIT_VAL 255
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/mechlovin/adelais/standard_led/avr/rev1/config.h:49]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mechlovin/adelais/rgb_led/rev3/config.h (hunk 3 above)
Comment (mechlovin) [reply keyboards/mechlovin/adelais/rgb_led/rev3/config.h:18]: added.
thank you.

**Comment by drashna** [comment]
Also, because this is a significant change, it should probably target develop.

**Comment by mechlovin** [comment]
Can it then target the master?
The VIA requires the PCB to be in the QMK master.

**Comment by mechlovin** [comment]
Thank you for your suggestion. If we target Develop now, can it be moved/targeted to Master later and how? VIA requires the PCB to be in QMK Master.

**Comment by drashna** [comment]
> Thank you for your suggestion. If we target Develop now, can it be moved/targeted to Master later and how? VIA requires the PCB to be in QMK Master.

Develop gets merged into master quarterly, so it will end up in master, eventually.  Unfortunately, the last cycle ended a week or so ago, so it will be around November for the next merge. 

That said, the requirement seems to be a soft one.

**Comment by mechlovin** [comment]
@drashna Thank you for clarifying, and targeted to Develop as suggested.

**Review by drashna — APPROVED** [review]

File: keyboards/mechlovin/adelais/rules.mk [hunk 10]
```diff
@@ -17,8 +17,10 @@ COMMAND_ENABLE = yes        # Commands for debug and configuration
 SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
 # if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
 NKRO_ENABLE = yes           # USB Nkey Rollover
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
```
Comment (fauxpark) [inline keyboards/mechlovin/adelais/rules.mk:21]:
```suggestion
```

File: keyboards/mechlovin/adelais/rules.mk [hunk 11]
```diff
@@ -17,8 +17,10 @@ COMMAND_ENABLE = yes        # Commands for debug and configuration
...
 AUDIO_ENABLE = no           # Audio output on port C6
```
Comment (fauxpark) [inline keyboards/mechlovin/adelais/rules.mk:22]:
```suggestion
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/mechlovin/adelais/rgb_led/rev2/rev2.c [hunk 12]
```diff
@@ -17,7 +17,7 @@
 #include "adelais.h"
 
 #ifdef RGB_MATRIX_ENABLE
-const is31_led __flash g_is31_leds[DRIVER_LED_TOTAL] = {
+const is31_led g_is31_leds[DRIVER_LED_TOTAL] = {
```
Comment (fauxpark) [inline keyboards/mechlovin/adelais/rgb_led/rev2/rev2.c:20]:
```suggestion
const is31_led __flash g_is31_leds[DRIVER_LED_TOTAL] = {
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mechlovin/adelais/rgb_led/rev2/config.h [hunk 13]
```diff
@@ -1,7 +1,7 @@
 #pragma once
 
 #define PRODUCT_ID   0xAEC2
-#define PRODUCT      Adelais En Ciel Rev2
+#define PRODUCT      Adelais En Ciel Rev2
```
Comment (fauxpark) [inline keyboards/mechlovin/adelais/rgb_led/rev2/config.h:4]:
```suggestion
#define PRODUCT      Adelais En Ciel Rev2
```

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**